crates/provider/src/utils.rs

```diff
@@ -102,7 +102,7 @@ fn estimate_priority_fee(rewards: &[Vec<u128>]) -> u128 {
 
 /// The default EIP-1559 fee estimator.
 ///
-/// Based on the work by [MetaMask](https://github.com/MetaMask/core/blob/main/packages/gas-fee-controller/src/fetchGasEstimatesViaEthFeeHistory/calculateGasFeeEstimatesForPriorityLevels.ts#L56);
+/// Based on the work by [MetaMask](https://github.com/MetaMask/core/blob/main/packages/gas-fee-controller/src/GasFeeController.ts);
 /// constants for "medium" priority level are used.
```

Comment:
Looks like the `eth_feeHistory` implementation was removed from MetaMask in https://github.com/MetaMask/core/pull/4210. CC @mattsse @klkvr for visibility.

In this case I think we should pin the commit instead of updating the file:

```suggestion
/// Based on the work by [MetaMask](https://github.com/MetaMask/core/blob/0fd4b397e7237f104d1c81579a0c4321624d076b/packages/gas-fee-controller/src/fetchGasEstimatesViaEthFeeHistory/calculateGasFeeEstimatesForPriorityLevels.ts#L56);
/// constants for "medium" priority level are used.
```